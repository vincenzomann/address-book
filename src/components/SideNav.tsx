import { navigate } from '@reach/router';
import React from 'react';
import { FaAddressBook, FaMapMarkedAlt } from 'react-icons/fa';

interface Props { }

const SideNav: React.FC<Props> = () => {
	return (
		<div id='sideNav'>
			<div id='logoItem'>
				<img className='logo' id='logo-mob' src="https://bequest.app/static/media/icon.3ab9eb70.png" alt="Bequest" />
				<img className='logo' id='logo-full' src="https://bequest.app/static/media/logo.e211df9f.png" alt="Bequest" />
			</div>
			<div id='navItems'>
				<div className='navItem' onClick={() => navigate('/address-book')}>
					<span className='icon'><FaAddressBook size='25' /></span>
					<span className='linkText'>Address Book</span>
				</div>
				<div className='navItem' onClick={() => navigate('/lookup')}>
					<span className='icon'><FaMapMarkedAlt size='25' /></span>
					<span className='linkText'>Lookup Address</span>
				</div>
			</div>
		</div>
	);
};

export default SideNav;
