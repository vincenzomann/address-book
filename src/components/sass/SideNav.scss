#sideNav {
	height: 100%;
	min-width: 72px;
	border-right: 1px solid lightGrey;
	text-align: center;
	@media only screen and (min-width: 700px) {
		min-width: 240px;
		text-align: left;
	}

	#sideContainer {
		position: sticky;
		top: 0;
	}

	#logoItem {
		padding: 1.5rem 1rem;

		.logo {
			height: 2.25rem;
		}
		#logo-full {
			display: none;
			@media only screen and (min-width: 700px) {
				display: inline;
			}
		}
		#logo-mob {
			display: inline;
			@media only screen and (min-width: 700px) {
				display: none;
			}
		}
	}

	#navItems {
		:hover {
			background-color: #f5f5f5;
			cursor: pointer;
			* {
				color: #009bc1;
			}
		}

		.navItem {
			padding: 1rem 1rem;
			display: flex;
			align-items: center;
			justify-content: center;
			@media only screen and (min-width: 700px) {
				justify-content: left;
			}
			.icon {
				color: grey;
				@media only screen and (min-width: 700px) {
					padding: 0 0.5rem;
				}
			}
			.linkText {
				text-decoration: none;
				font-size: small;
				display: none;
				padding: 0 1rem;
				@media only screen and (min-width: 700px) {
					display: inline;
				}
			}
		}
	}
}
