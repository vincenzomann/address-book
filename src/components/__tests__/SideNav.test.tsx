import React from 'react';
import { shallow } from 'enzyme';
import toJSON from 'enzyme-to-json';
import SideNav from '../SideNav';

test('matches snapshot', () => {
	const wrapper = shallow(<SideNav />);
	expect(toJSON(wrapper)).toMatchSnapshot();
});
