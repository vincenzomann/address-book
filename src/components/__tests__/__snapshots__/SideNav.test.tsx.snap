// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  id="sideNav"
>
  <div
    id="sideContainer"
  >
    <div
      id="logoItem"
    >
      <img
        alt="Bequest"
        className="logo"
        id="logo-mob"
        src="https://bequest.app/static/media/icon.3ab9eb70.png"
      />
      <img
        alt="Bequest"
        className="logo"
        id="logo-full"
        src="https://bequest.app/static/media/logo.e211df9f.png"
      />
    </div>
    <div
      id="navItems"
    >
      <div
        className="navItem"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <FaAddressBook
            size="25"
          />
        </span>
        <span
          className="linkText"
        >
          Address Book
        </span>
      </div>
      <div
        className="navItem"
        onClick={[Function]}
      >
        <span
          className="icon"
        >
          <HiUserAdd
            size="25"
          />
        </span>
        <span
          className="linkText"
        >
          Add Address
        </span>
      </div>
    </div>
  </div>
</div>
`;
